arwen/contexts/storage.go

```diff
@@ -154,18 +156,41 @@ func (context *storageContext) GetStorageUnmetered(key []byte) []byte {
 	return context.getStorageFromAddressUnmetered(context.address, key)
 }
 
+// enableStorageProtection will prevent writing to protected keys
+func (context *storageContext) enableStorageProtection() {
+	context.arwenStorageProtectionEnabled = true
+}
+
+// disableStorageProtection will prevent writing to protected keys
+func (context *storageContext) disableStorageProtection() {
+	context.arwenStorageProtectionEnabled = false
+}
+
+func (context *storageContext) isArwenProtectedKey(key []byte) bool {
+	return bytes.HasPrefix(key, []byte(arwen.ProtectedStoragePrefix))
+}
+
 func (context *storageContext) isElrondReservedKey(key []byte) bool {
 	return bytes.HasPrefix(key, context.elrondProtectedKeyPrefix)
 }
 
+func (context *storageContext) SetProtectedStorage(key []byte, value []byte) (arwen.StorageStatus, error) {
```

Comment:
Missing comment.

Reply:
Will add on a different branch, thank you.